pkgs/development/tools/mold/default.nix

```diff
@@ -1,25 +1,25 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, nix-update-script
+, testers
+
 , cmake
 , mimalloc
+, mold
 , ninja
-, openssl
 , zlib
```

Comment:
There is some choice to make about the dependencies that `mold` optionally bundles.

For example, see https://github.com/rui314/mold/commit/7f7a744ebb9875e77f862a12e2fb9ca48ac27e7e#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR124

> Find BLAKE3 cryptographic hash library. Just like zlib, if libblkae3.so # is not found, we compile a bundled one and statically-link it to mold.

It's the same with many other libaries (`zstd`, `tbb`, ...) and also `zlib`.
We give `zlib` here but not the others.
Shouldn't we remove `zlib` as well then in this case?

Both approaches are valid: Using `mold`'s bundled ones will make packaging easier, but using nixpkgs's ones will make patching security issues potentially better as it does not require relying on `mold` to make a new release.

In any case think we should mention this fact, and that it's done consciously for one of these reasons, in a comment so that it's clear to others.

Reply:
Thanks for bringing this to my attention.

I prefer using `nixpkgs`'s ones exactly for the reasons you've stated. I've added a commit that adds system shared libraries where possible. The intent to use the system libraries should now be clear to others due to the amount of libraries in `buildInputs` and due to the flags in `cmakeFlags`.

Note that the C implementation of the [BLAKE3](https://github.com/BLAKE3-team/BLAKE3) library is not packaged in Nixpkgs yet, only the [binary](https://github.com/NixOS/nixpkgs/blob/99ff0167add0a7416d5d9b0cf8f9dbf83092e8c6/pkgs/tools/security/b3sum/default.nix) is.

Also, I haven't touched the static libraries that `mold` uses ([`rustc-demangle`](https://github.com/NixOS/nixpkgs/blob/99ff0167add0a7416d5d9b0cf8f9dbf83092e8c6/pkgs/development/libraries/rustc-demangle/default.nix) and [`xxHash`](`https://github.com/NixOS/nixpkgs/blob/99ff0167add0a7416d5d9b0cf8f9dbf83092e8c6/pkgs/development/libraries/xxHash/default.nix`)). Those could be replaced as well, although I don't think they have security issues.